sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/CodeGenInternal/PatchModels/AzureApplicationInsightsParameterPatch.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Text.Json;
+using Azure.Core;
+
+namespace Azure.AI.MetricsAdvisor.Models
+{
+    internal partial class AzureApplicationInsightsParameterPatch
+    {
+        // Full qualification must be used so CodeGen skips the generation of this method.
+        // See: https://github.com/Azure/autorest.csharp/issues/793
+        void global::Azure.Core.IUtf8JsonSerializable.Write(Utf8JsonWriter writer)
+        {
+            writer.WriteStartObject();
+            writer.WriteNullStringValue("azureCloud", AzureCloud);
+            writer.WriteNullStringValue("applicationId", ApplicationId);
+            if (Optional.IsDefined(ApiKey))
+            {
+                writer.WritePropertyName("apiKey");
+                writer.WriteStringValue(ApiKey);
+            }
```

Comment:
We're not sending null for secrets, though. There's a good reason for this. The service does not return secrets to us, so this would happen:
```cs
var dataFeed = client.GetDataFeed(...);

// Here, dataFeed.DataSource.ApiKey is null because it was not returned by the service.

// Update some properties here

// This would send apiKey = null to the service, even though we didn't touch it.
client.UpdateDataFeed(dataFeed);
```

Reply:
and will that cause an error in the service? if this was allowed before, would it be better to add an entry to the changelog saying how this "bug" got fixed?

Reply:
> and will that cause an error in the service?

Correct. We get back an error message saying the secret is required.

> if this was allowed before, would it be better to add an entry to the changelog saying how this "bug" got fixed?

We were not overwriting the serialization method before, so we were not sending null. My point in this comment is "let's keep the ApiKey behaving as it already was. If we change it, it breaks".